logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/register/RegisterAccountUseCase.kt

```diff
@@ -0,0 +1,54 @@
+package com.wire.kalium.logic.feature.register
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.ServerConfig
+import com.wire.kalium.logic.data.register.RegisterAccountRepository
+import com.wire.kalium.network.api.user.register.RegisterResponse
+
+sealed class RegistrationParam(
+    firstName: String,
+    LastName: String,
+    val email: String,
+    val password: String,
+) {
+    val name: String
+
+    init {
+        name = "$firstName $LastName"
+    }
+
+    class PrivateAccount(
+        firstName: String,
+        LastName: String,
+        email: String,
+        password: String,
+        val emailActivationCode: String
+    ) : RegistrationParam(firstName, LastName, email, password)
+}
+
+class RegisterAccountUseCase(
+    private val registerAccountRepository: RegisterAccountRepository,
+) {
+    suspend operator fun invoke(param: RegistrationParam, serverConfig: ServerConfig): RegisterResult {
+        return when (param) {
+            is RegistrationParam.PrivateAccount -> {
+                with(param) {
+                    registerAccountRepository.registerWithEmail(email, emailActivationCode, name, password, serverConfig.apiBaseUrl)
+                }
+            }
+        }.fold(
+            {
+                RegisterResult.Failure.Generic(it)
+            }, {
+                // TODO: login user
```

Comment:
Considering Kalium is supposed to implement the behaviour of our flagship apps, I guess it makes sense to do this.

Do you plan on taking this TODO on another PR?

Reply:
This will be implemented in a different PR since I need to introduce some changes to the login and client register API